scripts/python/packages/parthenon_tools/parthenon_tools/movie2d.py

```diff
@@ -142,45 +200,97 @@ def plot_dump(
                 )
                 p.add_patch(rect)
 
-    plt.savefig(name, dpi=300)
-    plt.close()
+    fig.savefig(output_file, dpi=300)
+    plt.close(fig=fig)
+    logger.debug(f"Saved {time_title}s time-step to {output_file}")
 
 
 if __name__ == "__main__":
-    addPath()
+    # addPath()
+    ERROR_FLAG = False
     args = parser.parse_args()
-    field = args.field
-    files = args.files
-    components = [0, 0]
-    if (args.tc is not None) and (args.vc is not None):
+    logger.setLevel(args.log_level)
+
+    if args.tc and args.vc:
         raise ValueError(
             "Only one of --tensor-component and --vector-component should be set."
         )
+
+    if args.workers > 1:
+        logger.warning(
+            "Matplotlib is not multi-thread friendly. Read this for more details https://matplotlib.org/stable/users/faq/howto_faq.html#work-with-threads"
+        )
+        logger.warning(
+            "Try decreasing threads if you encounter any undefined behaviour"
+        )
+    # Create output director if does't exists
+    args.output_directory.mkdir(0o755, True, True)
+    logger.info(f"Total files to process: {len(args.files)}")
+
+    components = [0, 0]
     if args.tc is not None:
         components = args.tc
     if args.vc is not None:
         components = [0, args.vc]
-    dump_id = 0
-    debug_plot = False
-    for f in files:
-        data = read(f)
-        print(data)
-        q = data.Get(field, False, not debug_plot)
-        name = str(dump_id).rjust(4, "0") + ".png"
-        if debug_plot:
-            plot_dump(
-                data.xg,
-                data.yg,
-                q,
-                name,
-                True,
-                data.gid,
-                data.xig,
-                data.yig,
-                data.xeg,
-                data.yeg,
-                components,
+
+    _x = ProcessPoolExecutor if args.worker_type == "process" else ThreadPoolExecutor
+    with _x(max_workers=args.workers) as pool:
+        for dump_id, file_name in enumerate(args.files):
+            data = phdf(file_name)
+            if args.field not in data.Variables:
+                logger.error(
+                    f'No such field "{args.field}" in {file_name}. This will lead to stop further processing'
+                )
+                logger.info(f"Available fields: {data.Variables}")
+                ERROR_FLAG = True
+                break
+
+            q = data.Get(args.field, False, not args.debug)
+            stem = f"{args.prefix}_{str(dump_id).rjust(4, '0')}".strip()
+            stem = re.sub(r"^_", "", stem)
+            name = stem + ".png"
```

Comment:
Regex and auxiliary variable here seems unnecessarily fancy. Why not just do the following?
```suggestion
            name = "{}_{:04d}.png".format(args.prefix, dump_id).strip()
```
sometimes old-style string-formatting is more clear and concise than f-strings. (Also the underscore for the prefix is probably desirable.)